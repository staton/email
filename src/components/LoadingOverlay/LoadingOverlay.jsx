import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LoadingOverlayState from '../../enums/LoadingOverlay';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';

const propTypes = {
    loadingOverlayState: PropTypes.string.isRequired
};

const defaultProps = {
    onClick: () => {}
};

export class LoadingOverlay extends Component {

	render() {
        return (this.props.loadingOverlayState === LoadingOverlayState.None)
            ? null
            : (
                <div className={this.getClassName()}>
                    <LoadingSpinner />
                </div>
            );
    }

    /**
     * Gets the class name(s) for this component.
     * @returns {string} The class name(s).
     */
    getClassName() {
        let className = 'LoadingOverlay';

        if (this.props.loadingOverlayState === LoadingOverlayState.Fading)
            className += ' LoadingOverlay--fade';
        
        return className;
    }

}

LoadingOverlay.propTypes = propTypes;
LoadingOverlay.defaultProps = defaultProps;

export default LoadingOverlay;