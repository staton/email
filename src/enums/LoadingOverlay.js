class LoadingOverlay {

    get Fading() {
        return "FADING";
    }

    get None() {
        return "NONE";
    }

    get Visible() {
        return "VISIBLE";
    }

}

const LoadingOverlayState = new LoadingOverlay();

export default LoadingOverlayState;