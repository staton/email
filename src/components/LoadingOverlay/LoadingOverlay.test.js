import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import {LoadingOverlay} from './LoadingOverlay';
import LoadingOverlayState from '../../enums/LoadingOverlay';

describe('<LoadingOverlay />', () => {

    it('renders correctly when visible', () => {
        const wrapper = shallow(<LoadingOverlay loadingOverlayState={LoadingOverlayState.Visible} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders correctly when not visible', () => {
        const wrapper = shallow(<LoadingOverlay loadingOverlayState={LoadingOverlayState.None} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('contains the ..--fade class when fading', () => {
        const wrapper = shallow(<LoadingOverlay loadingOverlayState={LoadingOverlayState.Fading} />);
        expect(wrapper.find('.LoadingOverlay--fade').length).toEqual(1);
    });
    
});