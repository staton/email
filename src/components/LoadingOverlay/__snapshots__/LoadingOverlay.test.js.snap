// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingOverlay /> renders correctly when not visible 1`] = `""`;

exports[`<LoadingOverlay /> renders correctly when visible 1`] = `
<div
  className="LoadingOverlay"
>
  <LoadingSpinner />
</div>
`;
